crates/cli-flags/src/lib.rs

```diff
@@ -924,3 +961,144 @@ impl CommonOptions {
         Ok(())
     }
 }
+
+pub trait FromFile {
+    fn from_file<P: AsRef<Path>>(path: P) -> Result<Self>
+    where
+        Self: Sized;
+}
+
+impl FromFile for Config {
+    fn from_file<P: AsRef<Path>>(path: P) -> Result<Self> {
+        let file_contents = fs::read_to_string(path.as_ref())?;
+        let mut common_options: CommonOptions = toml::from_str(&file_contents)?;
+        common_options.config(None)
+    }
+}
```

Comment:
This might perhaps be best as an inherent method on `CommonOptions` as an alternative constructor? That would still require the caller call `.config(...)` but that seems like a reasonable tradeoff to me.

Reply:
I agree